lib/std/build.zig

```diff
@@ -1059,7 +1059,10 @@ pub const Builder = struct {
 };
 
 test "builder.findProgram compiles" {
+    // TODO: uncomment and fix the leak
+    // const builder = try Builder.create(std.testing.allocator, "zig", "zig-cache", "zig-cache");
```

Comment:
The only consumption site actually uses an arena. I think it makes more sense for the builder to own the arena since the existing `.create` and `.destroy` leaves a poor example of memory cleanup.

Reply:
Builder assumes arena everywhere. Any tests of that code should not be leak checking